libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1352,12 +1464,79 @@ bool MainChain::AddTip(IntBlockPtr const &block)
 {
   FETCH_LOCK(lock_);
 
+  if (enable_stutter_removal_)
```

Comment:
Yes, this is true and I agree the usage of this function is a bit weird. I added this here to be safe in case it was later called not just for genesis. Indeed, AddTip should not be called on something that is a stutter block. I will change this to an assert instead to ensure correct usage.